Public/Src/Engine/Scheduler/Fingerprints/PipFingerprinter.cs

```diff
@@ -260,7 +274,12 @@ protected virtual void AddWeakFingerprint(IFingerprinter fingerprinter, Process
             AddFileOutput(fingerprinter, "StandardError", process.StandardError);
             AddFileOutput(fingerprinter, "StandardOutput", process.StandardOutput);
 
-            fingerprinter.AddOrderIndependentCollection<FileArtifact, ReadOnlyArray<FileArtifact>>("Dependencies", process.Dependencies, (fp, f) => AddFileDependency(fp, f), m_expandedPathFileArtifactComparer);            
+            fingerprinter.AddOrderIndependentCollection<FileArtifact, ReadOnlyArray<FileArtifact>>("Dependencies", process.Dependencies, (fp, f) => {

```

Comment:
I think it would be cleaner to write this as
```csharp
var relevantDependencies = untrackedExecutable
    ? process.Dependencies.Except(new[] { process.Executable })
    : process.Dependencies;
fingerprinter.AddOrderIndependentCollection<FileArtifact, ReadOnlyArray<FileArtifact>>(
    "Dependencies", 
    relevantDependencies, 
    (fp, f) => AddFileDependency(fp, f), m_expandedPathFileArtifactComparer); 
```